nodejs/rest-api/Dockerfile

```diff
@@ -1,22 +1,38 @@
-FROM node:18-alpine
+# Stage 1: Build and run tests
+FROM node:18-alpine AS build
 
 # Create app directory
 WORKDIR /usr/src/app
 
-# Install app dependencies
-# A wildcard is used to ensure both package.json AND package-lock.json are copied
-# where available (npm@5+)
+# Copy package files and install dependencies
 COPY package*.json ./
-
 RUN npm install
 
-# Bundle app source
+# Copy the rest of the application source
+COPY . .
+
+# Run unit tests
+RUN npm run test
+
+# Stage 2: Create the production image
+FROM node:18-alpine
```

Comment:
Do we need to have a multi-stage build for this? We are not using any artifact from the builder image